mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreen.tsx

```diff
@@ -0,0 +1,748 @@
+        <ThemedView
+          dark={tailwind('bg-transparent')}
+          light={tailwind('bg-transparent')}
+          style={tailwind('flex-row w-full')}
+        >
+          <WalletTextInput
+            autoCapitalize='none'
+            editable={!isDisabled}
+            onChange={(e) => {
+              if (!isDisabled) {
+                if (onChangeFromAmount !== undefined) {
+                  onChangeFromAmount(e.nativeEvent.text)
+                } else {
+                  onChange(e)
+                }
+              }
+            }}
+            placeholder={isDisabled ? undefined : translate('screens/CompositeSwapScreen', 'Enter an amount')}
+            style={tailwind('flex-grow w-2/5')}
+            testID={`text_input_${controlName}`}
+            value={value}
+            displayClearButton={(value !== defaultValue) && !isDisabled}
+            onClearButtonPress={() => onChangeFromAmount?.(defaultValue)}
+            title={title}
+            inputType='numeric'
+          >
+            {
+              (enableMaxButton && onChangeFromAmount !== undefined) && (
+                <>
+                  <SetAmountButton
+                    amount={new BigNumber(maxAmount ?? '0')}
+                    onPress={onChangeFromAmount}
+                    type={AmountButtonTypes.half}
+                  />
+
+                  <SetAmountButton
+                    amount={new BigNumber(maxAmount ?? '0')}
+                    onPress={onChangeFromAmount}
+                    type={AmountButtonTypes.max}
+                  />
+                </>
+              )
+            }
+            {
+              !enableMaxButton && (
+                <>
+                  <Icon height={20} width={20} />
+                  <ThemedText style={tailwind('pl-2')}>
+                    {token.displaySymbol}
+                  </ThemedText>
+                </>
+              )
+            }
+          </WalletTextInput>
+        </ThemedView>
+      )}
+      rules={rules}
+    />
+  )
+}
+
+/**
+ * This function finds all the possible adjacent vertices of an undirected graph in the context of tokens and pairs
+ * @param tokens
+ * @param pairs
+ * @param tokenFrom
+ */
+function getAllPossibleSwapToTokens (allTokens: TokenState[], pairs: DexItem[], tokenFrom: string): BottomSheetToken[] {
+  const graph: GraphProps[] = pairs.map(pair => {
+    const graphItem: GraphProps = {
+      pairId: pair.data.id,
+      a: pair.data.tokenA.id,
+      b: pair.data.tokenB.id
+    }
+    return graphItem
+  })
+
+  const reachableNodes: string[] = []
+  const reachableNodeIds = new Set<string>([])
+  // Use Sets to reduce checks if item is unique
+  const visitedNodes = new Set<string>([])
+  const nodesToVisit = new Set<string>([tokenFrom])
+
+  while (nodesToVisit.size !== 0) {
+    const [token] = nodesToVisit // first item in a set
+    const adjacentNodes = getAdjacentVertices(token, graph)
+    if (adjacentNodes !== undefined) {
```

Comment:
this is always existing since you're passing a default value `[]` on your reduce method